modules/Bio/EnsEMBL/MicroRNA.pm

```diff
@@ -0,0 +1,188 @@
+=head1 LICENSE
+
+Copyright [1999-2015] Wellcome Trust Sanger Institute and the EMBL-European Bioinformatics Institute
+Copyright [2016-2019] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+
+=head1 CONTACT
+
+  Please email comments or questions to the public Ensembl
+  developers list at <http://lists.ensembl.org/mailman/listinfo/dev>.
+
+  Questions may also be sent to the Ensembl help desk at
+  <http://www.ensembl.org/Help/Contact>.
+
+=cut
+
+=head1 NAME
+
+Bio::EnsEMBL::MicroRNA - A class representing a microRNA product
+of a transcript
+
+=head1 DESCRIPTION
+
+TODO
+
+=head1 SYNOPSIS
+
+  my $miR = Bio::EnsEMBL::MicroRNA->new(
+    -SEQ_START => 36,
+    -SEQ_END   => 58
+  );
+
+  # Stable-ID setter
+  $miR->stable_id('ENSM00090210');
+
+  # Get start and end position in the precursor transcript
+  my $start = $miR->start();
+  my $end = $miR->end();
+
+=cut
+
+
+package Bio::EnsEMBL::MicroRNA;
+
+use vars qw($AUTOLOAD);
+use strict;
+use warnings;
+
+use Bio::EnsEMBL::Utils::Exception qw( throw warning );
+use Bio::EnsEMBL::Utils::Argument qw( rearrange );
+use Bio::EnsEMBL::Utils::Scalar qw( assert_ref wrap_array );
+use Scalar::Util qw(weaken);
+
+use Bio::EnsEMBL::RNAProduct;
+
+use parent qw(Bio::EnsEMBL::RNAProduct);
+
+
+=head2 new
+
+  Arg: [-ARM]         : which arm of the hairpin precursor this miRNA comes
+                        from. Returns 3 and 5 for 3' and 5', respectively.
+  Arg [...]           : Named arguments to superclass constructor
+                        (see Bio::EnsEMBL::RNAProduct)
+  Example    : my $miR = Bio::EnsEMBL::MicroRNA->new(
+                 -SEQ_START => 36,
+                 -SEQ_END   => 58,
+                 -ARM       => 3
+               );
+  Description: Constructor.  Creates a new MicroRNA object
+  Returntype : Bio::EnsEMBL::MicroRNA
+  Exceptions : throw if ARM value is out of bounds
+  Caller     : general
+  Status     : In Development
+
+=cut
+
+sub new { ## no critic (Subroutines::RequireArgUnpacking)
```

Comment:
Could you add a comment about why this is required to be ignored for future reference

Reply:
Unless I have missed something, each of my "no critic" declarations should now be accompanied by an explanation.